kit/src/components/embeds/link_embed/get_link_data.rs

```diff
@@ -0,0 +1,110 @@
+use std::error::Error;
+
+use reqwest::Url;
+use scraper::{Html, Selector};
+
+pub async fn get_image_data(document: Html, meta_selector: Selector) -> Option<String> {
+    let mut image = None;
+
+    for element in document.select(&meta_selector) {
+        let prop_attr_opt = element.value().attr("property");
+        let content_attr_opt = element.value().attr("content");
+
+        if let (Some(prop_attr), Some(content_attr)) = (prop_attr_opt, content_attr_opt) {
+            if prop_attr == "og:image" {
+                image = Some(content_attr.to_string());
+            }
+        }
+    }
+
+    image
+}
+
+pub async fn get_title_data(document: Html, meta_selector: Selector) -> String {
```

Comment:
No need to be async if nothing is awaiting. 

```suggestion
pub fn get_title_data(document: Html, meta_selector: Selector) -> String {
```